From eafbaae223910e7d73f1961d5aee8b274e388f4e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 10 Dec 2019 19:26:43 +0000
Subject: [PATCH 0904/1000] MANTIS 0055465 CUSS 1.5.1 Biometrics Interface

CUSS-1.5.BIOMETRICS.GP.XSD
  - add gp-parameter-name::materna.quality.*.weight
  - remove gp-parameter-name::materna.cuss.biometricToken
  - add association-data-tag
  - move gp-parameter-name::materna.cuss.associate.recentPassPosition to association-data-tag::materna.cuss.associate.recentPassPosition

CUSS.BIOMETRIC.XSD:
  - add commandType identify

OneIdTypes.xsd:
  - add idDocumentData::isGenuine



git-svn-id: svn://localhost/SelfServiceCommon/trunk@1508 90b65887-3827-0410-9a23-83215b262276
---
 .../idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD       | 125 ++++++++++++++++--
 .../Massai/idl/Cuss/CUSS.BIOMETRIC.XSD        |   5 +
 .../Massai/idl/Cuss/OneIdTypes.xsd            |   5 +
 3 files changed, 124 insertions(+), 11 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
index e82afee1..7c135fef 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
@@ -116,28 +116,118 @@
           <xs:documentation>Absolute calculated/required image quality. Only pictures above this value must be sent by the device and processed by KA</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
-      <!-- Namespace 'materna.compare' -->
-      <xs:enumeration value="materna.compare.minimumScore">
+      <xs:enumeration value="materna.quality.verticalPosition.weight">
         <xs:annotation>
-          <xs:documentation>The required minimum score to be treated as a match. Paramater is optional as camera has a private default value (applicable for function(s): verify; applicable for message type(s): request)</xs:documentation>
+          <xs:documentation></xs:documentation>
         </xs:annotation>
       </xs:enumeration>
-      <xs:enumeration value="materna.compare.resultScore">
+      <xs:enumeration value="materna.quality.horizontalPosition.weight">
         <xs:annotation>
-          <xs:documentation>The resulting score of an image comparision (applicable for function(s): verify; applicable for message type(s): response)</xs:documentation>
+          <xs:documentation></xs:documentation>
         </xs:annotation>
       </xs:enumeration>
-      <!-- Namespace 'materna.cuss'> -->
-      <xs:enumeration value="materna.cuss.associate.recentPassPosition">
+      <xs:enumeration value="materna.quality.headImageWidthRatio.weight">
         <xs:annotation>
-          <xs:documentation>Touch point that has been passed by passenger (e.g. required by NEC Face Authentication Service's passage history registration). Send via the CUSS 'associate' function</xs:documentation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.headImageHeightRatio.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.frontalFace.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.brightness.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.equalBrightness.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
         </xs:annotation>
       </xs:enumeration>
-      <xs:enumeration value="materna.cuss.biometricToken">
+      <xs:enumeration value="materna.quality.naturalSkinColor.weight">
         <xs:annotation>
-          <xs:documentation>Biometric token which was returned by biometrics provider during enrolment. Also be used for deletion (purge) of biometrics data for a customer</xs:documentation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.faceContrast.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <xs:enumeration value="materna.quality.flashReflection.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.eyesOpened.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.lookStraight.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.noRedEyes.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.mouthClosed.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.noHat.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.noHeavyFrame.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.noFrameOnEyes.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.noTintedGlasses.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.qualityValue.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.quality.absoluteQuality.weight">
+        <xs:annotation>
+          <xs:documentation></xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <!-- Namespace 'materna.compare' -->
+      <xs:enumeration value="materna.compare.minimumScore">
+        <xs:annotation>
+          <xs:documentation>The required minimum score to be treated as a match. Paramater is optional as camera has a private default value (applicable for function(s): verify; applicable for message type(s): request)</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <xs:enumeration value="materna.compare.resultScore">
+        <xs:annotation>
+          <xs:documentation>The resulting score of an image comparision (applicable for function(s): verify; applicable for message type(s): response)</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+      <!-- Namespace 'materna.cuss'> -->
       <xs:enumeration value="materna.cuss.response.statusMessage">
         <xs:annotation>
           <xs:documentation>Optional status message returned by biometrics provider during operation execution (e.g. 'Success to register facial image (703)' or 'Face of camera image and passport image is different (-709)')</xs:documentation>
@@ -150,4 +240,17 @@
       </xs:enumeration>
     </xs:restriction>
   </xs:simpleType>
-</xs:schema>
\ No newline at end of file
+  <xs:simpleType name="association-data-tag">
+    <xs:annotation>
+      <xs:documentation>MATERNA specific keys for association data values of CUSS-1.5.BIOMETRICS-20191007.xsd. It's not referenced by another schema but just a central 'documentation' of the parameters supported by the MATERNA CUSS platform.</xs:documentation>
+    </xs:annotation>
+    <xs:restriction base="xs:string">
+      <!-- Namespace 'materna.cuss'> -->
+      <xs:enumeration value="materna.cuss.associate.recentPassPosition">
+        <xs:annotation>
+          <xs:documentation>Touch point that has been passed by passenger (e.g. required by NEC Face Authentication Service's passage history registration). Send via the CUSS 'associate' function</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
+    </xs:restriction>
+  </xs:simpleType>
+</xs:schema>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
index e3049263..9ef444ce 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
@@ -76,6 +76,11 @@
                     <xsd:documentation>Verifies if a passenger is already enrolled by given boarding pass data or if the the given camera image and boarding pass data is matching the enrolled passenger (supported by NEC service handler only)</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
+            <xsd:enumeration value="identify">
+                <xsd:annotation>
+                    <xsd:documentation>Identifies a passenger via Service Handler interface. Pendant for MODI function 'getIdentifiedBiometricImage'. What passenger details are retured depends on the biometics backend</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
         </xsd:restriction>
     </xsd:simpleType>
 
diff --git a/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
index e97b194c..d47a79e1 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
+++ b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
@@ -141,6 +141,11 @@
                     </xsd:simpleContent>
                 </xsd:complexType>
             </xsd:element>
+            <xsd:element name="isGenuine" type="xsd:boolean" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Indicates if this document is genuine. Depends on device handler implementation if this element is supported</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
         </xsd:sequence>
     </xsd:complexType>
 
-- 
2.41.0.windows.1

